downloads/swan-lake-release-notes/swan-lake-alpha2/RELEASE_NOTE.md

```diff
@@ -2,41 +2,554 @@
+configurable string certPath = ?;
+
+service on new tcp:Listener(9002, secureSocket = {
+    certificate: {path: certPath},
+    privateKey: {path: keyPath},
+    protocol: {
+        name: "TLS",
+        versions: ["TLSv1.2", "TLSv1.1"]
+    },
+    ciphers: ["TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA"]
+}) {
+
+    isolated remote function onConnect(tcp:Caller caller) returns tcp:ConnectionService {
+        io:println("Client connected to secureEchoServer: ", caller.remotePort);
+        return new EchoService(caller);
+    }
+}
+
+service class EchoService {
+    tcp:Caller caller;
+
+    public isolated function init(Caller c) {
+        self.caller = c;
+    }
+
+    remote function onBytes(readonly & byte[] data) returns (readonly & byte[])|tcp:Error? {
+        io:println("Echo: ", 'string:fromBytes(data));
+        return data;
+    }
+}
+```
+
+##### UDP Module Improvements
+
+- The `sendDatagram` function will now send multiple datagrams if the size of the `byte[]` value provided as the `data` field of the datagram exceeds 8KB.
+
+- Returning `Datagram` from the `onDatagram` or `onBytes` remote methods also sends multiple datagrams if the size of the `byte[]` value provided (as the `data` field of the datagram) exceeds 8KB.
+
+##### Crypto Module Improvements
+
+Added support to decode private keys from `.key` files and public keys from `.cert` files and update the APIs for decoding private/public keys. This enables reading private/public keys from PEM files.
+
+##### JWT Module Improvements
+
+Extended the private key support for JWT signature generation and public cert support for JWT signature validation.
+
+#### Code to Cloud
+
+- The `Kubernetes.toml` file is renamed to `Cloud.toml`. 
+- The `--cloud=docker` build option is implemented. This will build the Dockerfile and Docker image without generating the Kubernetes artifacts.
+
 
 #### Developer Tools
-- 
 
+##### Language Server
+
+Implemented renaming support in the language server. Now, VSCode users are able to select a symbol and rename all occurrences within the same module as well as across modules.
+
+##### Debugger
+
+Added variable paging support. With this feature, the Ballerina variables, which contain a large number of child variables will be shown in a paged view in the debug variable presentation.
+
+##### Ballerina Shell REPL [EXPERIMENTAL]
+
+1. Fixed the REPL expression output to output the `toBalString()` result.
+2. Improved the REPL parser to support some partial snippets such as the example cases below.
+	  - Template strings will allow continuing on new lines.
+	  - The CLI will wait for more input if the last character was an operator.
+	  - The CLI will not wait for unclosed double quotes.
+3. Enable REPL to exit on `Ctrl+D`.
+
+#### Breaking Changes
+
+- Member access on a value of type `table` now returns `()` if the `table` does not contain a member with the specified key. Otherwise, the result is the member of the `table` with the given key.
```

Comment:
```suggestion
- Member access on a value of type `table` now returns `()` if the `table` does not contain a member with the specified key. Otherwise, the result is the member in the `table` value with the given key.
```